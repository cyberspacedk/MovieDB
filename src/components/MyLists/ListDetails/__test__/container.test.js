import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import ListDetailsContainer from '../container';

describe('Container: ListDetailsContainer', () => {
  const props = {
    match: {
      params: {
        id: 999,
      },
    },
  };
  const store = configureStore()({
    listDetails: {
      ids: [1],
      totalResults: 10,
      loading: false,
      error: false,
    },
    database: {
      lists: {
        1: { id: 1, name: 'Comedy' },
      },
      movies: {
        1: { id: 1 },
      },
    },
  });

  store.dispatch = jest.fn();

  const wrapper = shallow(<ListDetailsContainer store={store} {...props} />);
  const container = wrapper.dive().dive();
  const instance = container.instance();

  it('Snapshot: should match', () => {
    expect(container).toMatchSnapshot();
  });

  it('Check call lifeCycleMethod componentDidMount. Should dispatch 2 actions', () => {
    instance.componentDidMount();

    expect(store.dispatch).toHaveBeenNthCalledWith(1, {
      type: 'GET_CREATED_LIST_REQUEST',
      payload: 1,
    });

    expect(store.dispatch).toHaveBeenNthCalledWith(2, {
      type: 'GET_LIST_DETAILS_REQUEST',
      payload: 999,
    });
  });
});
