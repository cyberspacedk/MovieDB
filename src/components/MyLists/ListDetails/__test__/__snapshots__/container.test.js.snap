// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container: ListDetailsContainer Snapshot: should match 1`] = `
<ListDetails
  deleteMovieFromListRequest={[Function]}
  detailsList={
    Array [
      Object {
        "id": 1,
      },
    ]
  }
  empty={false}
  error={false}
  getCreatedListRequest={[Function]}
  getListDetailsRequest={[Function]}
  listName=""
  loading={false}
  match={
    Object {
      "params": Object {
        "id": 999,
      },
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "payload": 1,
              "type": "GET_CREATED_LIST_REQUEST",
            },
          ],
          Array [
            Object {
              "payload": 999,
              "type": "GET_LIST_DETAILS_REQUEST",
            },
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
